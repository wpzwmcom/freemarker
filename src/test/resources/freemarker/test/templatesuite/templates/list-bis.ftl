<#list ['a', 'b', 'c'] as x>
    ${x?index}: ${x} [<#list ['A', 'B', 'C'] as x>${x?index}:${x}<#sep>, </#list>]
</#list>

<#list ['a', 'b', 'c'] as i>
    ${i?index}: ${i} <#list ['A', 'B', 'C'] as j>${i?index}${i}/${j?index}${j}<#sep>, </#list>
</#list>

<#list ['a', 'b', 'c']><#items as x>${x?index}:${x}<#sep>, </#items></#list>
<#list ['a', 'b', 'c']><#items as x>${x?counter}. ${x};<#sep> </#items></#list>

<#list ['a', 'b', 'c'] as x>${x}<#if x?hasNext>, </#if></#list>
<#list ['a', 'b', 'c'] as x\-y>${x\-y}<#if x\-y?hasNext>, </#if></#list>

<#list ['a', 'b', 'c'] as x><#if x?isFirst>${x?capFirst}<#else>${x}</#if><#sep>, </#sep><#if x?isLast>.</#if></#list>

<#list ['a', 'b', 'c'] as x>${x?isOddItem?c}/${x?isEvenItem?c}<#sep> </#list>

<#list ['a', 'b', 'c'] as x>
    <td class="${x?itemParity}Row">${x}</td>
</#list>

<#list ['a', 'b', 'c'] as x>
    <td class="row${x?itemParityCap}">${x}</td>
</#list>

<#list ['a', 'b', 'c', 'd', 'e', 'f', 'g'] as x>
    <td class="${x?itemCycle('R', 'G', 'B')}">${x}</td>
</#list>
<@assertFails message="expects 1"><#list 1..1 as x>${x?itemCycle()}</#list></@>