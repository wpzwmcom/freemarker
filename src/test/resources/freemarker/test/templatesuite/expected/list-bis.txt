    0: a [0:A, 1:B, 2:C]
    1: b [0:A, 1:B, 2:C]
    2: c [0:A, 1:B, 2:C]

    0: a 0a/0A, 0a/1B, 0a/2C
    1: b 1b/0A, 1b/1B, 1b/2C
    2: c 2c/0A, 2c/1B, 2c/2C

0:a, 1:b, 2:c
1. a; 2. b; 3. c;

a, b, c
a, b, c

A, b, c.

true/false false/true true/false

    <td class="oddRow">a</td>
    <td class="evenRow">b</td>
    <td class="oddRow">c</td>

    <td class="rowOdd">a</td>
    <td class="rowEven">b</td>
    <td class="rowOdd">c</td>

    <td class="R">a</td>
    <td class="G">b</td>
    <td class="B">c</td>
    <td class="R">d</td>
    <td class="G">e</td>
    <td class="B">f</td>
    <td class="R">g</td>
